stdlib/Dates/src/accessors.jl

```diff
@@ -41,6 +43,47 @@ function week(days)
     return div(w, 28) + 1
 end
 
+"""
+Year defined as ISO year. The equivalent is
+`week(lastdayofyear(dt) - Day(3))`  as 28 december is always in the last week
+"""
+function weeksinyear(dt::DateTime)
+    firstday = firstdayofyear(dt)
+    lastday = lastdayofyear(dt)
+
+    if dayofweek(firstday) == 4 || dayofweek(lastday) == 4
+        return 53
+    end
+    return 52
+end
+
+"""
+Return current ISO year as defined in
+https://en.wikipedia.org/wiki/ISO_week_date
```

Comment:
```suggestion
Return the ISO year that contains `dt` (see https://en.wikipedia.org/wiki/ISO_week_date).
```

(I avoid the term "definition" because Wikipedia may be a bit too fluid to use explicitly as definition.)